aristanetworks/ansible-avd
# Fix(eos_config_deploy_cvp): avoid making CV_DEVICES becoming None

… devices

## Change Summary



## Related Issue(s)

Fixes #<ISSUE ID>

## Component(s) name

`arista.avd.eos_config_deploy_cvp`

## Proposed changes
in https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_config_deploy_cvp/templates/cvp-devices-v3.j2
if the device_filter does not match any devices inside the for loop `CVP_DEVICES` will be empty, so it'll become `None` and then on the ansible-cvp side we'll get that TypeError traceback mentioned in the #1293 

This will only set `CVP_DEVICES` to an empty list, and then validation should be done in the cv_devices_v3 module on ansible-cvp side

## How to test
To check if the list will become empty simply  print the `CVP_DEVICES` var in https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_config_deploy_cvp/tasks/v3/present.yml after L41
like this:
```
- name: "Print CVP_DEVICES_RESULTS"
  ansible.builtin.debug:
    var: CVP_DEVICES
```

which should result in

```
TASK [arista.avd.eos_config_deploy_cvp : Configure devices on AVD-CVP-1] **********************************************************
ok: [AVD-CVP-1]

TASK [arista.avd.eos_config_deploy_cvp : Print CVP_DEVICES_RESULTS] ***************************************************************
ok: [AVD-CVP-1] => {
    "CVP_DEVICES": []
}
```

## Checklist

### User Checklist


- N/A

### Repository Checklist
